pilot/pkg/ambient/controller/workloadcache.go

```diff
@@ -54,7 +56,19 @@ func initWorkloadCache(opts Options) *workloadCache {
 		_, hasType := o.GetLabels()[ambient.LabelType]
 		return hasType
 	})
-
+	if features.IncludeLocalWorkload {
+		wc.indexes[ambient.TypeWorkload].Insert(ambient.Workload{
+			UID:              "",
+			Name:             "local",
+			Namespace:        "default",
+			Labels:           map[string]string{ambient.LabelType: ambient.TypeWorkload},
+			ServiceAccount:   "default",
+			NodeName:         "local",
+			PodIP:            "127.0.0.1",
+			PodIPs:           nil,
+			WorkloadMetadata: ambient.WorkloadMetadata{},
+		})
+	}
 	opts.Client.KubeInformer().Core().V1().Pods().Informer().AddEventHandler(proxyHandler)
```

Comment:
Its not actually a new watch/informer, this is using a shared informer. So just a custom event handler.

But I do agree. I had some vague ideas on improving the whole pipeline here even more, but nothing concrete

Reply:
I can do a draft on master - I would like to also add the pod labels to the result, no need to send them on each connection or request unsecured if we can just get them from the source. I think this reverse map of (network, IP) -> (workload info) should be as efficient as possible, in particular since we want very low latency when pods go up and down. 

The existing pipeline in Istiod is optimized for large batch updates - this is a very low cost / low latency call we can handle
in near real time, we can likely bypass debounce and a lot of other mechanisms intended for huge configs distributed to all pods. 

I would assume ztunnel caches the result for some time and tries again - but will only ask for  pods for which active
connections are getting open, so a change will only need to be distributed to the smaller list of nodes that are 
actively using the pod. If a pod nobody is connected to is changed - no push will be sent. And the update will
be identically sent to all the nodes that need it. This can't use the old flow.

Reply:
BTW - my interest around this is also for the 'trusted network' model, where info about IP is used to derive the identity instead of using the cert. I suspect will also provide a better solution for telemetry.

Reply:
Agreed. Let's get this PR merged so ztunnel testing can proceed with the branch, and we can work on master - I think we can support both 'send all' and on-demand, and the set of data is almost the same for all cases.